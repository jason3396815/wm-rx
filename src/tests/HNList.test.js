import { h, Component } from 'preact';
import HNList from '../components/HNList';
import { shallow } from 'preact-render-spy';

describe.only('<HNList />', () => {
	const fetchMock = jest.fn(() => Promise.resolve({
		json: () => ([]),
	}));

	global.fetch = fetchMock;

	it('should a HNList of articles', () => {
		expect(<HNList endpoint="/news" page="1" />).toMatchSnapshot();
	});

	it('should have called fetch()', () => {
		const context = shallow(<HNList endpoint="/news" page="1" />);
		expect(fetchMock).toHaveBeenCalledWith("https://node-hnapi.herokuapp.com/news?page=1");
	});
});

