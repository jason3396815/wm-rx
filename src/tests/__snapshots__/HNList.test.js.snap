// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HNList /> should a HNList of articles 1`] = `
e {
  "attributes": Object {
    "endpoint": "/news",
    "page": "1",
  },
  "children": Array [],
  "key": undefined,
  "nodeName": [Function],
}
`;
